doc/read-the-docs-site/extensions-flags-pragmas.rst

```diff
@@ -0,0 +1,83 @@
+The ``Strict`` extension ensures that let bindings are also (by default) strict, for instance, evaluating
+``let x = 3 + 4 in 42`` evaluates ``3 + 4`` first, even though ``x`` is not used.
+
+Bang patterns and lazy patterns can be used to explicitly specify whether a let binding is strict or non-strict, as in ``let !x = 3 + 4 in 42`` (strict) and ``let ~x = 3 + 4 in 42`` (non-strict).
+At this time, it is not possible to make function applications non-strict: ``(\(~x) -> 42) (3 + 4)`` still evaluates ``3 + 4`` strictly.
+
+Making let bindings strict by default has the following advantages:
+
+* It makes let bindings and function applications semantically equivalent, e.g., ``let x = 3 + 4 in 42`` has the same semantics as ``(\x -> 42) (3 + 4)``.
+  This is what one would come to expect, as it is the case in most other programming languages, regardless of whether the language is strict or non-strict.
+* Untyped Plutus Core programs, which are compiled from Plutus Tx, are not evaluated lazily (unlike Haskell), that is, there is no memoization of the results of evaluated expressions.
+  Thus using non-strict bindings can cause an expression to be inadvertently evaluated for an unbounded number of times.
+  Consider ``let x = <expensive> in \y -> x + y``.
+  If ``x`` is non-strict, ``<expensive>`` will be evalutated every time ``\y -> x + y`` is applied to an argument, which means it can be evaluated 0 time, 1 time, 2 times, or any number of times (this is not the case if lazy evaluation was employed).
+  On the other hand, if ``x`` is strict, it is always evaluated once, which is at most one more time than what is necessary.
+
+Flags
+-------------------------------------------------
+
+GHC has a variety of optimization flags, many of which are on by default.
+Although Plutus Tx is, syntactically, a subset of Haskell, it has different semantics and a different evaluation strategy (Haskell: non-strict semantics, call by need; Plutus Tx: strict semantics, call by value).
+As a result, some GHC optimizations are not helpful for Plutus Tx programs, and can even be harmful, in the sense that it can make Plutus Tx programs less efficient, or fail to be compiled.
+An example is the full laziness optimization, controlled by GHC flag ``-ffull-laziness``, which floats let bindings out of lambdas whenever possible.
+Since Untyped Plutus Core does not employ lazy evaluation, the full laziness optimization is never beneficial, and can sometimes make a Plutus Tx program more expensive.
+Conversely, some GHC features must be turned on in order to ensure Plutus Tx programs are compiled successfully.
+
+All Plutus Tx modules should use the following GHC flags: ::
+
+  -fplugin PlutusTx.Plugin
+  -fno-ignore-interface-pragmas
+  -fno-omit-interface-pragmas
+  -fno-full-laziness
+  -fno-spec-constr
+  -fno-specialise
+  -fno-strictness
+  -fno-unbox-strict-fields
+  -fno-unbox-small-strict-fields
```

Comment:
You may want to specify how to add those (`.cabal` file vs `OPTIONS_GHC`). I know it's not the best place to do it, but it could be helpful. Or at least provide a reference.